### PR TITLE
Disable crossgen when poisoning

### LINKED ISSUE #3641: Crossgen breaking poison infra
Crossgen is modifying reference assemblies in SBRP by removing the poison marker we added to these files. The removal of the poison marker prevents the detection of leaked files.

### PR DESCRIPTION
**Pull request by ellahathaway**
Fixes https://github.com/dotnet/source-build/issues/3641

This PR introduces changes to conditionally disable crossgen during poisoning and alert the user via the UI that crossgen has been disabled.

### CODE REVIEW

File: src/SourceBuild/content/build.proj [hunk 1]
```diff
@@ -34,6 +34,7 @@
 
     <Message Text="Build Mode: $(BuildModeInfoText)" Importance="high" />
     <Message Text="Build Environment: $(TargetArchitecture) $(Configuration) $(TargetOS) $(TargetRid)" Importance="high" />
+    <Message Condition="'$(EnablePoison)' == 'true'" Text="Crossgen is disabled for poisoning" Importance="high" />
```
Comment (ViktorHofer) [inline src/SourceBuild/content/build.proj:37]: I would rephrase this slightly to something like "Crossgen is disabled because it's in-compatible when performing poison checks.".

Comment (ellahathaway) [reply src/SourceBuild/content/build.proj]: Addressed in https://github.com/dotnet/sdk/pull/42228/commits/f6cd9d3c9b3fd82df237577170d79b75bdf2cd59

**Review by ViktorHofer — APPROVED** [review]

File: src/SourceBuild/content/Directory.Build.props [hunk 2]
```diff
@@ -190,6 +190,7 @@
     <PoisonReportDataFile>$(PackageReportDir)poison-catalog.xml</PoisonReportDataFile>
     <PoisonedReportFile>$(PackageReportDir)poisoned.txt</PoisonedReportFile>
     <PoisonUsageReportFile>$(PackageReportDir)poison-usage.xml</PoisonUsageReportFile>
+    <DISABLE_CROSSGEN>true</DISABLE_CROSSGEN>
```
Comment (ViktorHofer) [inline src/SourceBuild/content/Directory.Build.props:193]: Does this work? I think this would need to be defined as an env var when building the sdk repository. But aren't there are also other repositories that invoke crossgen?

Comment (mthalman) [inline src/SourceBuild/content/Directory.Build.props:193]: Shouldn't this be conditioned on whether poisoning is enabled?

Comment (mthalman) [reply src/SourceBuild/content/Directory.Build.props:193]: Oh, never mind. It's in a `PropertyGroup` that is conditioned.

Comment (ellahathaway) [reply src/SourceBuild/content/Directory.Build.props:193]: I took this prop from [here](https://github.com/dotnet/dotnet/blob/67e54850463002d2502780a8a6058206d44c02be/src/sdk/src/Installer/redist-installer/targets/Crossgen.targets#L5) and [here](https://github.com/dotnet/dotnet/blob/67e54850463002d2502780a8a6058206d44c02be/src/sdk/eng/DotNetBuild.props#L18) in sdk. When this issue was first discovered, setting this property to true resulted in newly detected poison leaks.

A quick investigation shows that crossgen is used in [runtime](https://github.com/dotnet/dotnet/blob/e01b3fa3c490b78d0c87c48a2fe418e9d2d99407/src/runtime/src/tasks/Crossgen2Tasks/Microsoft.NET.CrossGen.targets#L134) (for example). I think the question I'm having trouble answering now is whether this is something that needs to be disabled. To me, https://github.com/dotnet/dotnet/blob/67e54850463002d2502780a8a6058206d44c02be/src/sdk/src/Installer/redist-installer/targets/Crossgen.targets#L5 is a clear example of crossgenning assemblies we use during the build. I can't say if the runtime example is doing the same?

Comment (ViktorHofer) [reply src/SourceBuild/content/Directory.Build.props:193]: OK. FWIW this property doesn't flow into dotnet/sdk. It's just set in the VMR orchestrator and never gets passed to dotnet/sdk's build.

Comment (ellahathaway) [reply src/SourceBuild/content/Directory.Build.props]: image
You were right about it not getting passed. I moved the prop to the sdk.proj file and now crossgen has been disabled as expected. Thanks for catching that.

**Review by ViktorHofer — APPROVED** [review]

**Review by mthalman — APPROVED** [review]
